circle.yml

```diff
@@ -2,11 +2,31 @@ machine:
   java:
     version:
       oraclejdk8
+  environment:
+    KAFKA_SERVER: 'localhost:9092'
 
 dependencies:
   override:
     - ./gradlew classes testClasses
 
+    # Download kafka if it isn't cached
+    - '[ -d ~/kafka_2.11-0.10.0.0 ] || curl -L http://apache.mirror.gtcomm.net/kafka/0.10.0.0/kafka_2.11-0.10.0.0.tgz | tar -xz -C ~/'
+
+    # Start Zookeeper in the background
+    - '~/kafka_2.11-0.10.0.0/bin/zookeeper-server-start.sh ~/kafka_2.11-0.10.0.0/config/zookeeper.properties':
+        background: true
+
+    # Start Kafka in the background
+    - '~/kafka_2.11-0.10.0.0/bin/kafka-server-start.sh ~/kafka_2.11-0.10.0.0/config/server.properties':
+        background: true
+
+    # Wait for Kafka to start
+    - sleep 10
```

Comment:
I'd pull in our script to wait for a socket so you don't have to guess on a wait time.